### PR TITLE
Azure Communication Service - Phone Number Administration - Live Test

### PR DESCRIPTION
**Pull request by ytyeason**
Moving the same approved PR in the azure-sdk-for-python-pr repo to this public repo: **https://github.com/Azure/azure-sdk-for-python-pr/pull/618**

I would appreciate if you guys could approve again :)

### CODE REVIEW

File: sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+    @AsyncCommunicationTestCase.await_prepared_test
+    @pytest.mark.live_test_only
+    async def test_list_all_phone_numbers(self):
+        async with self._phone_number_administration_client:
+            pages = self._phone_number_administration_client.list_all_phone_numbers()
+
+            items = []
+            async for item in pages:
+                items.append(item)
+        self.assertGreater(len(items), 0)
```
Comment (xiangyan99) [inline sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py:114]: you can just use assert len(items) > 0

File: sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    @AsyncCommunicationTestCase.await_prepared_test
+    @pytest.mark.live_test_only
+    async def test_get_all_area_codes(self):
+        async with self._phone_number_administration_client:
+            area_codes = await self._phone_number_administration_client.get_all_area_codes(
+                location_type="NotRequired",
+                country_code=self.country_code,
+                phone_plan_id=self.phone_plan_id_area_codes
+            )
+        assert area_codes.primary_area_codes is not None
```
Comment (xiangyan99) [inline sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py:125]: Given primary_area_codes is not boolean, you can just use 
assert area_codes.primary_area_codes

File: sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        pstnConfig = PstnConfiguration(
+            callback_url="https://callbackurl",
+            application_id="ApplicationId",
+            azure_pstn_target_id="AzurePstnTargetId"
+        )
+        async with self._phone_number_administration_client:
+            await self._phone_number_administration_client.configure_number(
+                pstn_configuration=pstnConfig,
+                phone_number=self.phonenumber_to_configure
+            )
```
Comment (xiangyan99) [inline sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py:183]: What do you want to verify?

File: sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+        assert search_response.search_id is not None
+
+    @AsyncCommunicationTestCase.await_prepared_test
+    @pytest.mark.live_test_only
+    async def test_cancel_search(self):
+        async with self._phone_number_administration_client:
+            await self._phone_number_administration_client.cancel_search(
+                search_id=self.search_id_to_cancel
+            )
```
Comment (xiangyan99) [inline sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py:277]: Same, is there a way to validate the result? (the search is successfully cancelled?)

Comment (ytyeason) [reply sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client_async.py:294]: Added asserts to functions returning None

File: sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client.py [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            self.search_id_to_cancel = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_SEARCH_ID_TO_CANCEL')
+            self.phonenumber_to_configure = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_PHONENUMBER_TO_CONFIGURE')
+            self.phonenumber_to_get_config = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_PHONENUMBER_TO_GET_CONFIG')
+            self.phonenumber_to_unconfigure = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_PHONENUMBER_TO_UNCONFIGURE')
+            self.phonenumber_for_capabilities = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_PHONENUMBER_FOR_CAPABILITIES')
+            self.capabilities_id = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_CAPABILITIES_ID')
+            self.release_id = os.getenv('AZURE_COMMUNICATION_SERVICE_TNM_RELEASE_ID')
+            self.scrubber.register_name_pair(
+                self.phone_plan_group_id,
+                "phone_plan_group_id"
```
Comment (xiangyan99) [inline sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client.py:53]: This does not match the one you specified in playback mode
self.phone_plan_group_id = "phone-plan-group-id-1"
Please make sure it works.

Comment (ytyeason) [reply sdk/communication/azure-communication-administration/tests/test_phone_number_administration_client.py:53]: matched the srubber string with the playback mode string. thanks

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**